src/pages/task/TaskBreadcrumb.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { Breadcrumb } from "antd";
+import { paths } from "contants/routes";
+
+interface Props {
+  version: string;
+  displayName: string;
+}
+
+export const TaskBreadcrumb: React.FC<Props> = ({ version, displayName }) => {
```

Comment:
I'm not seeing the need for 2 breadcrumb components. This one already contains the patch one embedded in it more or less, and having the props contain the breadcrumb trail as a list of { label : link } should solve the problem you mentioned initially.

Reply:
I explain why 2 is best implementation in summary but will give it another go:

1. Using one breadcrumb component requires rendering outside of any specific routes. This makes the React Router hooks unusable, which is required to render the correct breadcrumbs. 
2. One breadcrumb component means that the breadcrumb component needs to query for the task data if rendered on the task page. This already creates somewhat messy logic to pull off. Furthermore, the data it is querying is the same data queried for int he task page. However the data cannot be easily shared between the breadcrumb component and the task page. This is one of my least favorite parts of gql on the client so far--there's no easy way to subscribe to the same data like in Redux. The two queries could be grouped together although it would require complex configuration at the link level. As you can see, the code to achieve this is already becoming complex and hard to follow, making it prone to bugs.

The code that is shared between the two components is so low that it does not pose a material reason to converge them. When written this way they are incredibly straight forward with no conditional logic, create no extra queries, require no complex configuration at the link level, and are easy to test because there is no conditional logic.

As someone who tried for an hour and a half to pull off one breadcrumb component, I strongly think this is the best way to go.

Reply:
when i was thinking "one breadcrumb" i thought "one breadcrumb component rendered at the App level. Now I see what you were requesting, which is one breadcrumb component rendered in two places: Patch and Task. That I can do!